src/main/java/duke/logic/Duke.java

```diff
@@ -0,0 +1,58 @@
+package duke.logic;
+
+import java.io.IOException;
+
+import duke.logic.commands.Command;
+import duke.logic.commands.CommandResult;
+import duke.logic.parser.Parser;
+import duke.logic.tasks.TaskList;
+import duke.storage.Storage;
+
+/**
+ * The logic of the app, parsing user input and performing commands accordingly.
+ */
+public class Duke {
+    private Parser parser;
+    private TaskList taskList;
+
+    /**
+     * Creates an instance of the program logic that can parse user input
+     * and perform commands.
+     *
+     * @return a Duke instance serving as the program logic.
+     */
+    public static Duke init() {
+        Duke d = new Duke();
+        d.parser = new Parser();
+        try {
+            d.taskList = new TaskList(Storage.load());
+        } catch (IOException e) {
+            // todo Handle corrupt data
+            System.out.printf("Cannot load tasks\n  %s\nCreated a new task list.", e.getMessage());
+            d.taskList = new TaskList();
+        }
+        return d;
+    }
+
+    /**
+     * Processes the user input and obtains the response.
+     *
+     * @param userCommandText The user input.
+     * @return A string describing the program's response to the user input.
+     */
+    public String getResponse(String userCommandText) {
+        Command command = parser.parse(userCommandText);
+        command.setTaskList(taskList);
+        CommandResult result = command.execute();
+        return result.getMessage();
+    }
```

Comment:
Perhaps you could mention what this function returns in the summary of the method description?

I noticed a similar trend in some of the method descriptions. Is there a reason you prefer describing what the method is intended to be used for in another part of the code instead of what the method does and returns?

Reply:
I don't think the module's Java coding standard really specify what the header comment should contain (do correct me if I'm wrong!) so I followed how the Java API writes comment, which is just a short description of what the method does. And personally I think it would be verbose to include the return value, since there is a `@return` clause dedicated to that purpose already. But I suppose I can check with prof on this issue! Thanks for the input.